flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/optimize/program/FlinkRuntimeFilterProgramTest.java

```diff
@@ -202,6 +202,32 @@ public void testBuildSideIsJoinWithoutExchange() throws Exception {
         util.verifyPlan(query);
     }
 
+    @Test
+    public void testBuildSideIsLeftJoinWithoutExchange() throws Exception {
```

Comment:
I have one question, after reverting your change, the plan is as follows:
```
HashJoin(joinType=[InnerJoin], where=[=(amount, amount0)], select=[id, amount, price, fact_date_sk, id0, name, amount0, price0, fact_date_sk0, id00, male, amount00, price00, dim_date_sk], build=[left])
:- Exchange(distribution=[hash[amount]])
:  +- RuntimeFilter(select=[amount], estimatedFilterRatio=[0.99993896484375])
:     :- Exchange(distribution=[broadcast])
:     :  +- GlobalRuntimeFilterBuilder(select=[amount], estimatedRowCount=[65536], maxRowCount=[218454])
:     :     +- Exchange(distribution=[single])
:     :        +- LocalRuntimeFilterBuilder(select=[amount], estimatedRowCount=[65536], maxRowCount=[218454])
:     :           +- Calc(select=[id, male, amount, price, dim_date_sk], where=[<(price, 500)])
:     :              +- TableSourceScan(table=[[testCatalog, test_database, dim, filter=[]]], fields=[id, male, amount, price, dim_date_sk])
:     +- TableSourceScan(table=[[testCatalog, test_database, fact2]], fields=[id, amount, price, fact_date_sk])
+- HashJoin(joinType=[LeftOuterJoin], where=[=(amount0, amount)], select=[id, name, amount, price, fact_date_sk, id0, male, amount0, price0, dim_date_sk], build=[right])
   :- Exchange(distribution=[hash[amount]])
   :  +- TableSourceScan(table=[[testCatalog, test_database, fact]], fields=[id, name, amount, price, fact_date_sk])
   +- Exchange(distribution=[hash[amount]])
      +- Calc(select=[id, male, amount, price, dim_date_sk], where=[<(price, 500)])
         +- TableSourceScan(table=[[testCatalog, test_database, dim, filter=[]]], fields=[id, male, amount, price, dim_date_sk])
```
For this pattern, I think only injecting runtime filter for fact2 is also meaningful, but after your change, there doesn't exist runtime filter. So I think this test case doesn't cover your exchange.

Reply:
After an offline discussion with @lsyldliu , we think it is meaningful to directly inject runtime filter for fact2(let the output of the `LeftOuterJoin` as a builder), even if it will affect the creation of `MultipleInput`. We will keep it as a future optimization, no changes will be made in this version.